---
title: "Analyzing high-throughput data"
author: "Frederick J. Boehm"
date: "7/27/2020"
output: html_document
---

This example is taken from: http://genomicsclass.github.io/book/pages/confounding.html

A 2007 Nature Genetics article (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3005333/) 
found a high proportion of genes differentially expressed between two ethnic groups. 

They attributed these differences to ethnicity. 



```{r, eval = FALSE}
remotes::install_bioc("Biobase")
remotes::install_bioc("genefilter")
remotes::install_github("genomicsclass/GSE5859")
```



```{r}
library(Biobase) ##available from Bioconductor
library(genefilter) ## also on Bioconductor
library(GSE5859) ##available from github
data(GSE5859)
```

We can extract the gene expression data and sample information table using the Bioconductor functions exprs and pData like this:

```{r}
geneExpression = exprs(e)
sampleInfo = pData(e)
```


Note that some samples were processed at different times.

```{r}
head(sampleInfo$date)
```

This is an extraneous variable and should not affect the values in geneExpression. However, as we have seen in previous analyses, it does appear to have an effect. We will therefore explore this here.

We can immediately see that year and ethnicity are almost completely confounded:


```{r}
year = factor( format(sampleInfo$date,"%y") )
tab = table(year,sampleInfo$ethnicity)
print(tab)
```

By running a t-test and creating a volcano plot, we note that thousands of genes appear to be differentially expressed between ethnicities. Yet when we perform a similar comparison only on the CEU population between the years 2002 and 2003, we again obtain thousands of differentially expressed genes:

```{r}
library(genefilter)

##remove control genes
out <- grep("AFFX",rownames(geneExpression))

eth <- sampleInfo$ethnicity
ind<- which(eth%in%c("CEU","ASN"))
res1 <- rowttests(geneExpression[-out,ind],droplevels(eth[ind]))
ind <- which(year%in%c("02","03") & eth=="CEU")
res2 <- rowttests(geneExpression[-out,ind],droplevels(year[ind]))

XLIM <- max(abs(c(res1$dm,res2$dm)))*c(-1,1)
YLIM <- range(-log10(c(res1$p,res2$p)))
plot(res1$dm,-log10(res1$p),xlim=XLIM,ylim=YLIM,
     xlab="Effect size",ylab="-log10(p-value)",main="Populations")
plot(res2$dm,-log10(res2$p),xlim=XLIM,ylim=YLIM,
     xlab="Effect size",ylab="-log10(p-value)",main="2003 v 2002")
```


